pkg/controllers/management/clusterprovisioner/provisioner.go

```diff
@@ -291,19 +300,70 @@ func (p *Provisioner) reconcileCluster(cluster *v3.Cluster, create bool) (*v3.Cl
 	return cluster, nil
 }
 
-func skipProvisioning(cluster *v3.Cluster) bool {
-	return cluster.Status.Driver == v3.ClusterDriverLocal || cluster.Status.Driver == v3.ClusterDriverImported
+func copyMap(toCopy v3.MapStringInterface) v3.MapStringInterface {
+	newMap := v3.MapStringInterface{}
+
+	for k, v := range toCopy {
+		newMap[k] = v
+	}
+
+	return newMap
 }
 
-func (p *Provisioner) getConfig(reconcileRKE bool, spec v3.ClusterSpec, driverName, clusterName string) (*v3.ClusterSpec, interface{}, error) {
-	data, err := convert.EncodeToMap(spec)
+func (p *Provisioner) censorGenericEngineConfig(input v3.ClusterSpec) (v3.ClusterSpec, error) {
+	if input.GenericEngineConfig == nil {
+		// nothing to do
+		return input, nil
+	}
+
+	config := copyMap(*input.GenericEngineConfig)
+	driverName, ok := config[DriverNameField].(string)
+	if !ok {
+		// can't figure out driver type so blank out the whole thing
+		logrus.Warnf("cluster %v has a generic engine config but no driver type field; can't hide password "+
+			"fields so removing the entire config", input.DisplayName)
+		input.GenericEngineConfig = nil
+		return input, nil
+	}
+
+	driver, err := p.KontainerDriverLister.Get("", driverName)
+	if err != nil {
+		return v3.ClusterSpec{}, err
+	}
+
+	var schemaName string
+	if driver.Spec.BuiltIn {
+		schemaName = driver.Status.DisplayName + "Config"
+	} else {
+		schemaName = driver.Status.DisplayName + "EngineConfig"
+	}
+
+	kontainerDriverSchema, err := p.DynamicSchemasLister.Get("", strings.ToLower(schemaName))
 	if err != nil {
-		return nil, nil, err
+		return v3.ClusterSpec{}, fmt.Errorf("error getting dynamic schema %v", err)
+	}
+
+	for key := range config {
+		field := kontainerDriverSchema.Spec.ResourceFields[key]
+		if field.Type == "password" {
+			delete(config, key)
```

Comment:
so you dropped password.... then where does the actual password like gke auth get stored?

Reply:
The `censorGenericEngineConfig` func is called in 2 places.
1) Line:276 to prevent the password fields from being put in the applied spec
2) Line:480 when comparing applied specs to see if a cluster needs to be reconciled.
So the password fields will still get stored in the `genericEngineConfig` field on the cluster, all this does is prevent it from appearing in other places where Norman won't automatically hide the fields.